pkg/secretsmanager/secretsmanager_integration_test.go

```diff
@@ -148,3 +156,76 @@ func TestEnsureGCPSecrets(t *testing.T) {
 		t.Fatalf("Expected 'admin', got: '%s'", data)
 	}
 }
+
+// newAzureClient create an Azure client with an authorizer from the environment
+func newCliAzureClient() (*keyvault.BaseClient, error) {
+	authorizer, err := azauth.NewAuthorizerFromCLI()
+	if err != nil {
+		return &keyvault.BaseClient{}, err
+	}
+	client := keyvault.New()
+	client.Authorizer = authorizer
+	return &client, nil
+}
+
+// TestLoadAzure requires the vault to be pre-configured at the moment and for hard deletes too
+// az keyvault create --name "secret-agent-test" --resource-group "secret-agent-test" --location eastus --enable-soft-delete false
+func TestLoadAzure(t *testing.T) {
+	keyConfig := &types.KeyConfig{
+		Name: "username",
+	}
+	secretConfig := &types.SecretConfig{
+		Name:      "load-azure-secrets",
+		Namespace: "default",
+		Keys:      []*types.KeyConfig{keyConfig},
+	}
+	node := &types.Node{
+		Path:         []string{"load-azure-secrets", "username"},
+		SecretConfig: secretConfig,
+	}
+	nodes := []*types.Node{node}
+
+	vaultName := "secret-agent-test"
+	// expect nothing found
+	baseCtx := context.Background()
+	ctx1, cancel := context.WithTimeout(baseCtx, 3000000000)
+	defer cancel()
+	client, err := newCliAzureClient()
+	err = loadAzureSecrets(ctx1, client, vaultName, nodes)
+	if err != nil {
+		t.Fatalf("Expected no error, got: %+v", err)
+	}
+	if len(node.Value) != 0 {
+		t.Fatalf("Expected no value, got: %+v", node.Value)
+	}
+
+	// create a secret to load
+	var secParams keyvault.SecretSetParameters
+	secName := getSecretID(node.SecretConfig.Namespace, node.Path)
+	secValue := "test-value"
+	secParams.Value = &secValue
```

Comment:
Maybe it's because I haven't had enough coffee yet, or because often glance over multiple lines at once instead of always reading linearly, but at first glance I read `secParams` as "security params". In my mind full var names would be easier on the eye to read, like `secretName`, `secretValue`, and `secretparams`. But that's just me. :)

Reply:
yea that's not a good convention, I've updated.